module/fastgltf.cppm

```diff
@@ -0,0 +1,162 @@
+module;
+
+#include <cassert>
+#ifndef FASTGLTF_USE_STD_MODULE
+#include <fastgltf/core.hpp>
+#endif
+
+export module fastgltf;
+
+#ifdef FASTGLTF_USE_STD_MODULE
+import std;
+import std.compat;
+
+extern "C++" {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winclude-angled-in-module-purview"
+#include <fastgltf/core.hpp>
+#pragma clang diagnostic pop
+}
+#endif
```

Comment:
What is the reason for these clang diagnostics? Looking at [this issue](https://github.com/llvm/llvm-project/issues/68615), it doesn't look like we should be including the header before the `export module` declaration.

Reply:
I think use `#include` in `extern "C++" { ... }` clause is okay to do. [This is from Clang documentation](https://clang.llvm.org/docs/StandardCPlusPlusModules.html#export-extern-c-style).

The reason I include the header file in export declaration region is: If `FASTGLTF_USE_STD_MODULE` macro defined, the definition of `std::`-prefixed things should be known before `<fastgltf/core.hpp>` header inclusion, however I can't place `import std;` inside the global module fragment, such like:

```c++
module;

import std; // Can't be here!
#include <fastgltf/core.hpp>

export module fastgltf;

// ...
```

Therefore, if `FASTGLTF_USE_STD_MODULE` macro not defined (and consequently not using `std` module), header inclusion should be inside the global module fragment. But otherwise, it have to be in export declaration regions, wrapped with `extern "C++" { ... }` clause.

Reply:
And I take it the `import std` is a C++23 feature? And using that has compile time benefits? Sorry, I'm quite unfamiliar with how modules work exactly. Though I guess this is fine.

Reply:
> And I take it the import std is a C++23 feature?

`import std` is C++23 feature [(P2465R3)](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p2465r3.pdf), but [major compiler vendors approved to support the feature in C++20](https://github.com/microsoft/STL/issues/3945) (I think it's because it is easier to adopt `std` module than header units).

> And using that has compile time benefits?

Yes, module compiled only once, therefore `import fastgltf;` doesn't introduce any compile time overhead. Also it is likely to user already compiled `std` module, so reusing it is reasonable.